packages/i18n/src/types.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Return type for string translation functions.
+ *
+ * Used for type checking of sprintf().
+ */
+export type TranslatableText< T extends string > = string & {
+	readonly __translatableText: T;
+};
```

Comment:
I would feel better if we made it very clear that this should just be `string`. I tried using some tricks to hide the phantom property but couldn't make it work:

```suggestion
/**
 * Return type for string translation functions.
 *
 * This type should be treated as if it were `string`.
 */
export type TranslatableText< T extends string > = string & {
	/**
	 * DO NOT USE! This property _does not exist_.
	 * @private
	 */
	readonly __translatableText: T;
};
```